docs/csharp/roslyn-sdk/work-with-semantics.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Work with the .NET Compiler SDK semantic model
+description: This overview provides an understanding of the type you use to understand and manipulate the semantic model of your code.
+author: billwagner
+ms.author: wiwagn
+ms.date: 10/15/2017
+ms.topic: conceptual
+ms.prod: .net
+ms.devlang: devlang-csharp
+ms.custom: mvc
+---
+
+# Work with Semantics
+
+Syntax trees represent the lexical and syntactic structure of source code. Although this information alone is enough to describe all the declarations and logic in the source, it is not enough information to identify what is being referenced. For those tasks, you need the **Semantic model**.
+
+For example, many types, fields, methods, and local variables with the same name may be spread throughout the source. Although each of these is uniquely different, determining which one an identifier actually refers to often requires a deep understanding of the language rules. 
+
+There are program elements represented in source code, and programs can also refer to previously compiled libraries, packaged in assembly files. Although no source code is available for assemblies and therefore no syntax nodes or trees, programs can still refer to elements inside them.
+
+In addition to a syntactic model of the source code, a semantic model encapsulates the language rules, giving you an easy way to make these distinctions.
+
+## Compilation
+
+A compilation is a representation of everything needed to compile a C# or Visual Basic program, which includes all the assembly references, compiler options, and source files. 
+
+Because all this information is in one place, the elements contained in the source code can be described in more detail. The compilation represents each declared type, member, or variable as a symbol. The compilation contains a variety of methods that help you find and relate the symbols that have either been declared in the source code or imported as metadata from an assembly.
+
+Similar to syntax trees, compilations are immutable. After you create a compilation, it cannot be changed by you or anyone else you might be sharing it with. However, you can create a new compilation from an existing compilation, specifying a change as you do so. For example, you might create a compilation that is the same in every way as an existing compilation, except it may include an additional source file or assembly reference.
+
+## Symbols
+
+A symbol represents a distinct element declared by the source code or imported from an assembly as metadata. Every namespace, type, method, property, field, event, parameter, or local variable is represented by a symbol. 
+
+A variety of methods and properties on the <xref:Microsoft.CodeAnalysis.Compilation> type help you find symbols. For example, you can find a symbol for a declared type by its common metadata name. You can also access the entire symbol table as a tree of symbols rooted by the global namespace.
```

Comment:
Do you want the xref to be the FQN instead of just type name? If so, add ?displayProperty=nameWithType

Reply:
I did mean the type name here. I thought it read more clearly. This type is abstract, and developers will use one of the derived classes (VB or C# specific).